mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -0,0 +1,94 @@
+from azure.storage.blob import BlobClient
+
+import os
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.utils.string_utils import strip_suffix
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+from mlflow.utils.rest_utils import call_endpoint, extract_api_info_for_service
+from mlflow.protos.databricks_artifacts_pb2 import DatabricksMlflowArtifactsService
+from mlflow.protos.databricks_artifacts_pb2 import GetCredentialsForWrite, GetCredentialsForRead
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.protos.service_pb2 import MlflowService, ListArtifacts
+
+_PATH_PREFIX = "/api/2.0"
+
+
+class DatabricksArtifactRepository(ArtifactRepository):
+    """
+    SOMETHING : TYPING TILL IT WORKS LOL
+    """
+
+    def __init__(self, artifact_uri):
+        super(DatabricksArtifactRepository, self).__init__(artifact_uri)
+
+    def _extract_run_id(self, artifact_uri):
+        return artifact_uri.lstrip('/').split('/')[4]
+
+    def _call_endpoint(self, service, api, json_body):
+        _METHOD_TO_INFO = extract_api_info_for_service(service, _PATH_PREFIX)
+        endpoint, method = _METHOD_TO_INFO[api]
+        response_proto = api.Response()
+        return call_endpoint(get_databricks_host_creds(), endpoint, method, json_body, response_proto)
+
+    def _create_json_body(self, run_id, path=None):
+        path = path or '.'
+        return {
+            "run_id": run_id,
+            "path": path
+        }
+
+    def _get_azure_write_credentials(self, run_id, path=None):
```

Comment:
This method calls an MLflow Artifacts Service endpoint that is not directly bound to Azure; it may also return AWS credentials. The type of credential is specified by the `type` field of the `ArtifactCredentialInfo` response. Accordingly, I think we should call this `_get_write_credentials`.

Callers of this method should check the `type` field of the response and then perform the corresponding cloud operation (either AWS or Azure). The current task is to implement Azure uploads / downloads, so it's fine to leave the AWS case unimplemented (e.g., simply raise a "Not Implemented" exception in the code). It would still be great to structure the upload / download logic so that it accounts for the fact that credentials may be provided for different cloud services. Let me know if this makes sense!

Reply:
Done.